Documentation/Troubleshooting/ceph-common-issues.md

```diff
@@ -507,6 +507,75 @@ You don't need to restart the pod, the effect will be immediate.
 
 To disable the logging on file, simply set `log_to_file` to `false`.
 
+## Collect perf data of a ceph process at runtime
+
+There are some cases where the debug logs are not sufficient to investigate issues
+like high CPU utilization of a Ceph process. In that situation, coredump and perf information
+of a Ceph process is useful to be collected which can be shared with the Ceph team in an issue.
+
+To collect this information, please follow these steps:
+
+* Edit the rook-ceph-operator deployment and set `ROOK_HOSTPATH_REQUIRES_PRIVILEGED` to `true`.
+* Wait for the pods to get reinitialized:
+
+```console
+# watch kubectl -n rook-ceph get pods
+```
+
+* Enter the respective pod of the Ceph process which needs to be investigated. For example:
+
+```console
+# kubectl -n rook-ceph exec -it deploy/rook-ceph-mon-a -- bash
+```
+
+* Install `gdb` , `perf` and `git` inside the pod. For example:
+```console
+# dnf install gdb git perf -y
+```
+
+* Capture perf data of the respective Ceph process:
+
+```console
+# perf record -e cycles --call-graph dwarf -p <pid of the process>
+# perf report > perf_report_<process/thread>
+```
+
+* Grab the `pid` of the respective Ceph process to collect its backtrace at multiple time instances, attach `gdb` to it and share the output `gdb.txt`:
+
+```console
+# gdb -p <pid_of_the_process>
+
+- set pag off
+- set log on
+- thr a a bt full #this captures the complete backtrace of the process
+- backtrace
+- Ctrl+C
+- backtrace
+- Ctrl+C
+- backtrace
+- Ctrl+C
+- backtrace
```

Comment:
Why are there multiple bactrace and Ctrl+C commands?

Reply:
Hi @travisn in one of troubleshooting instances recently Radek from RADOS team suggested collecting the continuous backtrace this way, when we press Ctrl+C there's a Quit call in gdb that happens:
```
(gdb) backtrace
#0  0x00007f6f54b1144c in pthread_cond_wait@@GLIBC_2.3.2 () from target:/lib64/libpthread.so.0
#1  0x00007f6f541588f0 in std::condition_variable::wait(std::unique_lock<std::mutex>&) () from target:/lib64/libstdc++.so.6
#2  0x00007f6f570ce61f in AsyncMessenger::wait() () from target:/usr/lib64/ceph/libceph-common.so.2
#3  0x0000556e0bccf56a in main ()
(gdb) Quit
(gdb) backtrace
#0  0x00007f6f54b1144c in pthread_cond_wait@@GLIBC_2.3.2 () from target:/lib64/libpthread.so.0
#1  0x00007f6f541588f0 in std::condition_variable::wait(std::unique_lock<std::mutex>&) () from target:/lib64/libstdc++.so.6
#2  0x00007f6f570ce61f in AsyncMessenger::wait() () from target:/usr/lib64/ceph/libceph-common.so.2
#3  0x0000556e0bccf56a in main ()
(gdb) Quit
(gdb) backtrace
#0  0x00007f6f54b1144c in pthread_cond_wait@@GLIBC_2.3.2 () from target:/lib64/libpthread.so.0
#1  0x00007f6f541588f0 in std::condition_variable::wait(std::unique_lock<std::mutex>&) () from target:/lib64/libstdc++.so.6
#2  0x00007f6f570ce61f in AsyncMessenger::wait() () from target:/usr/lib64/ceph/libceph-common.so.2
#3  0x0000556e0bccf56a in main ()
```

Reply:
I think we can have it here based on this.

Reply:
Sounds good to keep it, maybe a brief comment in the doc would help the users understand too. Otherwise it still feels like doing a magical dance. :)

Reply:
@travisn sure added.